pulser/_seq_drawer.py

```diff
@@ -132,11 +138,42 @@ def phase_str(phi):
                 unit = 'ns' if time_scale == 1 else r'$\mu s$'
                 ax.set_xlabel(f't ({unit})', fontsize=12)
 
+    if sampling_rate:
+        # Solver_time = np.arange(seq._total_duration, dtype=np.double)/1000
+        indexes = np.linspace(0, seq._total_duration-1,
+                              int(sampling_rate*seq._total_duration),
+                              dtype=int)
+        times = np.arange(seq._total_duration, dtype=np.double)/time_scale
+        solver_time = times[indexes]
+        delta_t = np.diff(solver_time)[0]
+        # Compare pulse with an interpolated pulse with 100 times more samples
+        teff = np.arange(0, max(solver_time), delta_t/100)
+
+        cs_amp = {}
+        cs_detuning = {}
+
     for ch, (a, b) in ch_axes.items():
         basis = seq._channels[ch].basis
         t = np.array(data[ch]['time']) / time_scale
         ya = data[ch]['amp']
         yb = data[ch]['detuning']
+        if sampling_rate:
+            nt = len(t)
+            t2 = 1
+            ya2 = []
+            yb2 = []
+            for t_solv in solver_time:
+                while t_solv > t[t2]:
+                    t2 += 1
+                    if t2+1 < nt:
+                        if t[t2+1] == t[t2]:
+                            t2 += 1
+                ya2.append(ya[t2])
+                yb2.append(yb[t2])
```

Comment:
This block is a bit confusing to me, I don't really get why you're not just taking the values at `t_solv`. Can you elaborate on that, please? Maybe add a comment in the code too

Reply:
This dates back to when there was a double counting at the jonction between two pulses. You've corrected it since then. I've just simply removed lines 168-170 now. I guess it's now clearer.